tests/system/providers/google/cloud/vertex_ai/example_vertex_ai_auto_ml.py

```diff
@@ -0,0 +1,688 @@
+    )
+    delete_bucket = GCSDeleteBucketOperator(
+        task_id="delete_bucket", bucket_name=FORECAST_GCS_BUCKET_NAME, trigger_rule=TriggerRule.ALL_DONE
+    )
+
+    clear_folder = BashOperator(
+        task_id="clear_forecast_folder",
+        bash_command="rm -r /forecast/*",
+    )
+
+    (
+        # TEST SETUP
+        create_bucket
+        >> unzip_file
+        >> upload_files
+        >> create_forecast_dataset
+        # TEST BODY
+        >> create_auto_ml_forecasting_training_job
+        # TEST TEARDOWN
+        >> delete_auto_ml_forecasting_training_job
+        >> delete_forecast_dataset
+        >> delete_bucket
+        >> clear_folder
+    )
+
+
+with models.DAG(
+    f"{DAG_ID}_image_training_job",
+    schedule_interval="@once",
+    start_date=datetime(2021, 1, 1),
+    catchup=False,
+    tags=["example", "vertex_ai", "auto_ml"],
+) as image_training_job_dag:
+    create_bucket = GCSCreateBucketOperator(
+        task_id="create_bucket",
+        bucket_name=IMAGE_GCS_BUCKET_NAME,
+        storage_class="REGIONAL",
+        location=REGION,
+    )
+
+    unzip_file = BashOperator(
+        task_id="unzip_csv_data_file", bash_command=f"unzip {IMAGE_ZIP_CSV_FILE_LOCAL_PATH} -d /image/"
+    )
+
+    upload_files = LocalFilesystemToGCSOperator(
+        task_id="upload_file_to_bucket",
+        src=IMAGE_CSV_FILE_LOCAL_PATH,
+        dst=IMAGE_GCS_OBJECT_NAME,
+        bucket=IMAGE_GCS_BUCKET_NAME,
+    )
+
+    create_image_dataset = CreateDatasetOperator(
+        task_id="image_dataset",
+        dataset=IMAGE_DATASET,
+        region=REGION,
+        project_id=PROJECT_ID,
+    )
+    image_dataset_id = create_image_dataset.output['dataset_id']
+
+    import_image_dataset = ImportDataOperator(
+        task_id="import_image_data",
+        dataset_id=image_dataset_id,
+        region=REGION,
+        project_id=PROJECT_ID,
+        import_configs=IMAGE_DATA_CONFIG,
+    )
+
+    # [START how_to_cloud_vertex_ai_create_auto_ml_image_training_job_operator]
+    create_auto_ml_image_training_job = CreateAutoMLImageTrainingJobOperator(
+        task_id="auto_ml_image_task",
+        display_name=f"auto-ml-image-{ENV_ID}",
+        dataset_id=image_dataset_id,
+        prediction_type="classification",
+        multi_label=False,
+        model_type="CLOUD",
+        training_fraction_split=0.6,
+        validation_fraction_split=0.2,
+        test_fraction_split=0.2,
+        budget_milli_node_hours=8000,
+        model_display_name=f"auto-ml-image-model-{ENV_ID}",
+        disable_early_stopping=False,
+        region=REGION,
+        project_id=PROJECT_ID,
+    )
+    # [END how_to_cloud_vertex_ai_create_auto_ml_image_training_job_operator]
+
+    delete_auto_ml_image_training_job = DeleteAutoMLTrainingJobOperator(
+        task_id="delete_auto_ml_training_job",
+        training_pipeline_id=create_auto_ml_image_training_job.output['training_id'],
+        region=REGION,
+        project_id=PROJECT_ID,
```

Comment:
Do we need trigger_rule for it or DeleteDatasetOperator removes the whole resource anyway? (I don't know the service specifics)

Reply:
Yes, it is needed. I have added this trigger_rule to all other delete tasks as well :)